apps/studio/components/interfaces/Settings/Database/ConnectionPooling/ConnectionPooling.tsx

```diff
@@ -169,22 +169,32 @@ export const ConnectionPooling = () => {
       max_client_conn: null,
     },
   })
-  const { type, default_pool_size, max_client_conn } = form.watch()
+  const { type, default_pool_size } = form.watch()
   const error = useMemo(
-    () => (type === 'PgBouncer' ? pgbouncerConfigError : supavisorConfigError),
-    [type]
+    () =>
+      allowPgBouncerSelection && type === 'PgBouncer' ? pgbouncerConfigError : supavisorConfigError,
+    [type, allowPgBouncerSelection]
```

Comment:
You need to add `pgbouncerConfigError` and `supavisorConfigError` as deps here, otherwise the value won't change when an error happens.

Same for the rest of the useMemos.

Reply:
ahh okay gotcha on this! 🙏